src/compiler/transformers/module/module.ts

```diff
@@ -1613,16 +1625,27 @@ export function transformModule(context: TransformationContext): (x: SourceFile
             for (const specifier of node.exportClause.elements) {
                 const exportNeedsImportDefault = !!getESModuleInterop(compilerOptions) &&
                     !(getInternalEmitFlags(node) & InternalEmitFlags.NeverApplyImportHelper) &&
-                    idText(specifier.propertyName || specifier.name) === "default";
-                const exportedValue = factory.createPropertyAccessExpression(
-                    exportNeedsImportDefault ? emitHelpers().createImportDefaultHelper(generatedName) : generatedName,
-                    specifier.propertyName || specifier.name,
-                );
+                    moduleExportNameIsDefault(specifier.propertyName || specifier.name);
+                const specifierName = specifier.propertyName || specifier.name;
+                const exportedValue = specifierName.kind === SyntaxKind.StringLiteral
+                    ? factory.createElementAccessExpression(
+                        exportNeedsImportDefault ? emitHelpers().createImportDefaultHelper(generatedName) : generatedName,

```

Comment:
Since this is the same as the expression on L1636, if you extract this line out to its own variable and use it in both branches then this statement won't require quite as much nesting.